romancal/outlier_detection/outlier_detection_step.py

```diff
@@ -1,37 +1,202 @@
-"""
-Detect and flag outlier in a science image
-"""
+"""Public common step definition for OutlierDetection processing."""
 
-import roman_datamodels as rdm
+import contextlib
+import os
+from functools import partial
+
+from roman_datamodels import datamodels
+
+from romancal.datamodels import ModelContainer
+from romancal.outlier_detection import outlier_detection
 
 from ..stpipe import RomanStep
 
+# Categorize all supported versions of outlier_detection
+outlier_registry = {
+    "imaging": outlier_detection.OutlierDetection,
+}
+
 __all__ = ["OutlierDetectionStep"]
 
 
 class OutlierDetectionStep(RomanStep):
-    """Detect and flag outliers in a science image."""
+    """Flag outlier bad pixels and cosmic rays in DQ array of each input image.
+
+    Input images can be listed in an input association file or already opened
+    with a ModelContainer.  DQ arrays are modified in place.
+
+    Parameters
+    -----------
+    input_data : asn file or ~romancal.datamodels.ModelContainer
+        Single filename association table, or a datamodels.ModelContainer.
+
+    """
+
+    class_alias = "outlier_detection"
+
+    # The members of spec needs to be a super-set of all parameters needed
+    # by the various versions of the outlier_detection algorithms, and each
+    # version will pick and choose what they need while ignoring the rest.
+    spec = """
+        weight_type = option('ivm','exptime',default='ivm')
+        pixfrac = float(default=1.0)
+        kernel = string(default='square') # drizzle kernel
+        fillval = string(default='INDEF')
+        nlow = integer(default=0)
+        nhigh = integer(default=0)
+        maskpt = float(default=0.7)
+        grow = integer(default=1)
+        snr = string(default='5.0 4.0')
+        scale = string(default='1.2 0.7')
+        backg = float(default=0.0)
+        kernel_size = string(default='7 7')
+        threshold_percent = float(default=99.8)
+        ifu_second_check = boolean(default=False)
+        save_intermediate_results = boolean(default=False)
+        resample_data = boolean(default=True)
+        good_bits = string(default="~DO_NOT_USE")  # DQ flags to allow
+        scale_detection = boolean(default=False)
+        search_output_file = boolean(default=False)
+        allowed_memory = float(default=None)  # Fraction of memory to use for the combined image
+        in_memory = boolean(default=False)
```

Comment:
We need to get the definition of all parameters. Also for jwst. Perhaps @stsci-hack can help with this.

Reply:
Will do. Thanks, Nadia!